dfget/core/downloader/p2p_downloader/client_writer.go

```diff
@@ -295,3 +295,38 @@ func sendSuccessPiece(api api.SupernodeAPI, cid string, piece *Piece, cost time.
 			piece.DstCid, cost.Seconds(), piece.Range)
 	}
 }
+
+func sendSuccessAlreadyDownloadedPiece(api api.SupernodeAPI, node string, taskID string, cid string, dstCid string, pieceRange string, notifyQueue queue.Queue) {
```

Comment:
Thanks for your contribution. 

And could you please make a refactor for function `sendSuccessAlreadyDownloadedPiece` and `sendSuccessPiece` to avoid too many identical codes.

Reply:
Yes, it has been merged now, thanks for reminding. @Starnop